graphene/types/field.py

```diff
@@ -121,18 +121,22 @@ def wrap_resolve(self, parent_resolver):
         """
         Wraps a function resolver, using the ObjectType resolve_{FIELD_NAME}
         (parent_resolver) if the Field definition has no resolver.
+
+        The Field resolver cannot be asynchronous generator.
         """
         if self.get_resolver is not None:
             warn_deprecation(
                 "The get_resolver method is being deprecated, please rename it to wrap_resolve."
             )
             return self.get_resolver(parent_resolver)
 
-        return self.resolver or parent_resolver
+        return not inspect.isasyncgenfunction(self.resolver) and self.resolver or parent_resolver
 
```

Comment:
Why do we need to change this to include the inspection? A field can either be used for subscriptions or for normal queries (XOR). This also might break future support of `stream` or `defer` depending on how they'll be integrated